### DISCUSSION TITLE
[SOLUTION] How to run on Ubuntu 20.x (Tips, not question - readme if you have an error)

### DISCUSSION THREAD

**Discussion by 503Dev** [sn4k3/UVtools#375] [Tutorials]
## Ubuntu 20.x and other releases will not run UVtools by default, you must install an extra package to make this work on a stock install.
TLDR; `sudo apt-get install -y libgeotiff-dev`

## Test your install by:

1. Download the UVTools Linux zip you wish to use, it's precompiled
2. Extract the zip file 
3. Execute the "UVtools" executable `./UVtools`   (note the command line -c convert flag is also an easy way to test)

## Error Fix
You will likely have gotten an error such as this one:
`libcvextern: cannot open shared object file: No such file or directory`

You can actually debug this further with this command:
`ldd libcvextern.so`

You will see a missing dependency, for Ubuntu it should be **libgeotiff.so.5 => not found**

## The Fix
To fix this simply run the following:
`sudo apt-get install -y libgeotiff-dev`

Once that finishes you can use UVtools happily.

**Comment by sn4k3** [comment]
This is already an requirement stated and given here: https://github.com/sn4k3/UVtools#linux

**Reply by 503Dev** [reply]
Thank you! I am an idiot and somehow did not see that doc, I swear I am not that idiotic.  I am actually a dev too and I did search, not sure how the heck I missed that.  Googling the error + uvtools did not help.  Sorry for spamming your issues with a repeat and thanks for the absolutely awesome tool.

P.D. I greatly appreciate the command line support for basic operations like convert, any chance of adding some type of a progress output with stdout (maybe with an option -p or -v flag to keep it clean for normal users).  I know I could fork and add it but just offering a suggestion, either way thanks again, you sir, are amazing.

**Reply by sn4k3** [reply]
> Thank you! I am an idiot and somehow did not see that doc, I swear I am not that idiotic. I am actually a dev too and I did search, not sure how the heck I missed that. Googling the error + uvtools did not help. Sorry for spamming your issues with a repeat and thanks for the absolutely awesome tool.

There aren't many posts about UVtools and many users are Windows and Mac, so google will fail to query that. All info are in Github readme and Wikis.

> P.D. I greatly appreciate the command line support for basic operations like convert, any chance of adding some type of a progress output with stdout (maybe with an option -p or -v flag to keep it clean for normal users). I know I could fork and add it but just offering a suggestion, either way thanks again, you sir, are amazing.

No, the UVtools is an UI application, it cant offer console output or native std. It just parse the args and do the operations before any UI spawn and quit the process, it is not bound to console.
It require a new project made for Console Application, it existed before but i drop it (UVtoolsCmd -- https://github.com/sn4k3/UVtools/tree/master/UVtools.Cmd)
Maybe a good challenge for who want to complete it ;)

**Reply by 503Dev** [reply]
Thank you for the details.  I would consider it but your code is far nicer than my coding style, I may fork and attempt some wrappers for basic commands :)
